keras_nlp/applications/bert.py

```diff
@@ -66,118 +67,118 @@ class BertModel(keras.Model):
+        dropout=0.1,
         max_sequence_length=512,
         type_vocab_size=2,
-        **kwargs,
+        **kwargs
     ):
-        super().__init__(**kwargs)
 
-        self.vocab_size = vocab_size
-        self.hidden_size = hidden_size
-        self.num_layers = num_layers
-        self.num_attention_heads = num_attention_heads
-        self.max_sequence_length = max_sequence_length
-        self.type_vocab_size = type_vocab_size
-        self.inner_size = inner_size
-        self.inner_activation = keras.activations.get(inner_activation)
-        self.initializer_range = initializer_range
-        self.initializer = keras.initializers.TruncatedNormal(
+        # Create lambda functions from input params
+        inner_activation_fn = keras.activations.get(inner_activation)
+        initializer_fn = keras.initializers.TruncatedNormal(
             stddev=initializer_range
         )
-        self.dropout = dropout
 
-        self._embedding_layer = keras.layers.Embedding(
-            input_dim=vocab_size,
-            output_dim=hidden_size,
-            embeddings_initializer=self.initializer,
-            name="word_embeddings",
+        # Functional version of model
+        token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="input_ids"
+        )
+        segment_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="segment_ids"
+        )
+        # TODO(jbischof): improve handling of masking following
+        # https://www.tensorflow.org/guide/keras/masking_and_padding
+        input_mask = keras.Input(
+            shape=(None,), dtype="int32", name="input_mask"
         )
 
-        self._position_embedding_layer = keras_nlp.layers.PositionEmbedding(
-            initializer=self.initializer,
+        # Embed tokens, positions, and segment ids.
+        token_embedding = keras.layers.Embedding(
+            input_dim=vocab_size,
+            output_dim=hidden_size,
+            name=TOKEN_EMBEDDING_LAYER_NAME,
+        )(token_id_input)
+        position_embedding = keras_nlp.layers.PositionEmbedding(
+            initializer=initializer_fn,
             sequence_length=max_sequence_length,
             name="position_embedding",
-        )
-
-        self._type_embedding_layer = keras.layers.Embedding(
+        )(token_embedding)
+        segment_embedding = keras.layers.Embedding(
             input_dim=type_vocab_size,
             output_dim=hidden_size,
-            embeddings_initializer=self.initializer,
-            name="type_embeddings",
-        )
-
-        self._embedding_norm_layer = keras.layers.LayerNormalization(
+            name="segment_embedding",
+        )(segment_id_input)
+
+        # Sum, normailze and apply dropout to embeddings.
+        x = keras.layers.Add(
+            name="embedding_sum",
+        )((token_embedding, position_embedding, segment_embedding))
+        x = keras.layers.LayerNormalization(
             name="embeddings/layer_norm",
```

Comment:
we have a few different styles of naming here. We name to make these uniform, and I think the correct thing to do might be to use `get_uid` to make sure are names are unique. @fchollet is that correct?

See https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/densenet.py#L80

Reply:
Yes, we should not have 2 layers with the same name in the same model. So any framework component that creates new layers should leverage `get_uid()` to ensure name unification, unless we know it will only be called once per model.

Reply:
Right now each layer would only be called once unless someone created a model with two encoders, which I've never seen before.